usm_wrappers/salt_wrapper.py

```diff
@@ -0,0 +1,453 @@
+    '''
+    mon_key_path = cluster_dir + '/mon.key'
+    admin_key_path = cluster_dir + '/client.admin.keyring'
+    mon_map_path = cluster_dir + '/mon.map'
+
+    utils.execCmd([_ceph_authtool.cmd, '--create-keyring', mon_key_path,
+                   '--gen-key', '-n', 'mon.', '--cap', 'mon', 'allow *'])
+
+    utils.execCmd([_ceph_authtool.cmd, '--create-keyring', admin_key_path,
+                   '--gen-key', '-n', 'client.admin', '--set-uid=0', '--cap',
+                   'mon', 'allow *', '--cap', 'osd', 'allow *', '--cap',
+                   'mds', 'allow'])
+
+    utils.execCmd([_ceph_authtool.cmd, mon_key_path, '--import-keyring',
+                   admin_key_path])
+
+    cmd = [_monmaptool.cmd, '--create', '--clobber']
+    for m, v in monitors.iteritems():
+        cmd += ['--add', 'mon.' + m, v['address']]
+    cmd += ['--fsid', fsid, mon_map_path]
+    utils.execCmd(cmd)
+
+    return True
+
+
+def _get_mon_id_map(unused_mon_ids, minions):
+    mon_id_map = dict(zip(unused_mon_ids, minions))
+    monitors = {}
+    for id, minion in mon_id_map.iteritems():
+        monitors[id] = {
+            'name': minions[minion].get('monitor_name',
+                                        utils.get_short_hostname(minion)),
+            'address': minions[minion]['public_ip']
+        }
+    return mon_id_map, monitors
+
+
+def _add_ceph_mon_pillar_data(mon_id_map, cluster_name, monitors):
+    pillar_data = {}
+    for id, minion in mon_id_map.iteritems():
+        pillar_data[minion] = {'cluster_name': cluster_name, 'mon_id': id,
+                               'mon_name': monitors[id]['name'],
+                               'public_ip': monitors[id]['address']}
+    return pillar_data
+
+
+def setup_ceph_cluster(cluster_name, fsid, minions):
+    '''
+    :: cluster_name = STRING
+    :: fsid = UUID
+    :: minions = {MINION_ID: {'public_ip': IP_ADDRESS,
+                              'cluster_ip': IP_ADDRESS,
+                              'monitor_name': NAME}, ...}
+    '''
+    public_network, cluster_network = check_minion_networks(minions)
+    mon_id_map, monitors = _get_mon_id_map(_MON_ID_LIST, minions)
+
+    cluster_dir = _CEPH_CLUSTER_CONF_DIR + '/' + cluster_name
+    if not os.path.exists(cluster_dir):
+        os.makedirs(cluster_dir)
+
+    conf_file = cluster_dir + '/' + cluster_name + '.conf'
+    _gen_ceph_cluster_conf(conf_file, cluster_name, fsid, monitors,
+                           public_network)
+
+    _gen_keys(cluster_name, fsid, monitors, cluster_dir)
+
+    pillar_data = _add_ceph_mon_pillar_data(mon_id_map, cluster_name, monitors)
+    pillar = {'usm': pillar_data}
+
+    bootstrapped_minion = None
+    for id, minion in mon_id_map.iteritems():
+        out = run_state(local, minion, 'add_ceph_mon',
+                        kwarg={'pillar':
+                               {'usm': {'mon_bootstrap': True,
+                                        minion: pillar_data[minion]}}})
+        if not out:
```

Comment:
We are going to change this logic ie setup_ceph_cluster() will expect only one mon and usm core will call add_ceph_mon() for rest of mons to have more control on error handling